### PR TITLE
doc: 📝 updated public docs

### PR DESCRIPTION
**Pull request by mighty840**
## Motivation and Context

Updated the public facing documentation

## Summary

- Added instructions for building locally without a devcontainer
- Updated architecture description to match current status
- Fixed a typo 🤞 

## Checklist

Please ensure that your PR meets the following requirements:

### Documentation

- [X] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] ~Added a CHANGELOG entry for major changes.~

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

**Review by hauju — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,9 +83,49 @@ For more details on using dev containers, refer to the [official documentation](
...
 By using the dev container, you ensure a consistent development environment across all contributors. Happy coding!
 
+## Local development requirements (without devcontainer)
+
+### Building from source: Dependencies
+
+To build the ETOPay SDK from source, you will need to install the following dependencies in addition to rustlang:
+
+- `clang`: The C compiler
+- `llvm`: The LLVM compiler infrastructure
```
Comment (antbern) [inline CONTRIBUTING.md:93]: Just a small nitpick: do we really need llvm? Whenever I tried this on the Linux laptop I only needed to install `clang` and `lld`, `llvm` already comes as part of `rustc` if I remember correctly 🚀

File: docs/docs/Choosing ETOPay/Architecture.md [hunk 2]
```diff
@@ -12,13 +12,13 @@ The ETOPay services are a bundle of services, which work together, using various
...
 
-Currently, on the first request, a staging environment of a fresh ETOPay services instance is deployed. This allows the developers to test their application flow and integration with ETOPay before moving to production. Requesting a production instance is an extra step which needs to be done additionally. This helps the developers to fix a version for their production, while they continue testing new features and releases from ETOPay team on the staging environment, before requesting to update to production.
+Currently, on the first request, a test and a production environment of a fresh ETOPay services instance is deployed. The test instance allows the developers to test their application flow and integration with ETOPay before moving to production.
```
Comment (antbern) [inline docs/docs/Choosing ETOPay/Architecture.md:21]: Should we use `test` and `live` here to stay consistent?
```suggestion
Currently, on the first request, a test and a live environment of a fresh ETOPay services instance is deployed. The test instance allows the developers to test their application flow and integration with ETOPay before moving to production.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (antbern) [reply CONTRIBUTING.md]: I see it is still there in the commands for installation 🌵

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -82,9 +83,48 @@ For more details on using dev containers, refer to the [official documentation](
...
+```
+
+#### macOS (with Homebrew)
+
+To install the dependencies on macOS using Homebrew, run the following commands:
+
+```bash
+brew install clang llvm lld protobuf
+```
+
```
Comment (antbern) [inline CONTRIBUTING.md:120]: Need to remove from here too:
```suggestion
To install the dependencies on Ubuntu or Debian, run the following commands:

```bash
sudo apt update
sudo apt install clang lld protobuf-compiler
```

#### Fedora

To install the dependencies on Fedora, run the following commands:

```bash
sudo dnf install clang lld protobuf-compiler
```

#### macOS (with Homebrew)

To install the dependencies on macOS using Homebrew, run the following commands:

```bash
brew install clang lld protobuf
```

```

**Review by antbern — APPROVED** [review]
